lab_1/main.py

```diff
@@ -1,94 +1,121 @@
+        tokens.append(token)
+    return tokens
 
+
+# получение стоп-слов
+def get_stop_words() -> list:
+    with open("stop_words.txt") as file:
+        stop_words = file.readlines()
+    return stop_words
+
+
+# удаление стоп-слов
 def remove_stop_words(tokens: list, stop_words: list) -> list:
-    """
-    Removes stop words
-    :param tokens: a list of tokens
-    :param stop_words: a list of stop words
-    :return: a list of tokens without stop words
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy']
-    stop_words = ['the', 'is']
-    --> ['weather', 'sunny', 'man', 'happy']
-    """
-    pass
+    if checker(tokens, list):
+        tokens_clean = [token for token in tokens if token not in stop_words]
+        return tokens_clean
+    else:
+        return []
 
 
+# подсчет частотности
 def calculate_frequencies(tokens: list) -> dict:
-    """
-    Calculates frequencies of given tokens
-    :param tokens: a list of tokens without stop words
-    :return: a dictionary with frequencies
-    e.g. tokens = ['weather', 'sunny', 'man', 'happy']
-    --> {'weather': 1, 'sunny': 1, 'man': 1, 'happy': 1}
-    """
-    pass
+    if checker(tokens, list):
+        frequencies = {token: tokens.count(token) for token in tokens if type(token) is str}
+        return frequencies
+    else:
+        return {}
 
 
+# получение top-n популярных слов
 def get_top_n_words(freq_dict: dict, top_n: int) -> list:
-    """
-    Returns the most common words
-    :param freq_dict: a dictionary with frequencies
-    :param top_n: a number of the most common words to return
-    :return: a list of the most common words
-    e.g. tokens = ['weather', 'sunny', 'man', 'happy', 'and', 'dog', 'happy']
-    top_n = 1
-    --> ['happy']
-    """
-    pass
+    freq = list(freq_dict.items())
+    freq_s = sorted(freq, key=lambda num: num[1], reverse=True)
 
+    top = [freq_s[i][0] for i in range(top_n)]
+    return top
 
-def get_concordance(tokens: list, word: str, left_context_size: int, right_context_size: int) -> list:
-    """
-    Gets a concordance of a word
-    A concordance is a listing of each occurrence of a word in a text,
-    presented with the words surrounding it
-    :param tokens: a list of tokens
-    :param word: a word-base for a concordance
-    :param left_context_size: the number of words in the left context
-    :param right_context_size: the number of words in the right context
-    :return: a concordance
-    e.g. tokens = ['the', 'weather', 'is', 'sunny', 'the', 'man', 'is', 'happy',
-                    'the', 'dog', 'is', 'happy', 'but', 'the', 'cat', 'is', 'sad']
-    word = 'happy'
-    left_context_size = 2
-    right_context_size = 3
-    --> [['man', 'is', 'happy', 'the', 'dog', 'is'], ['dog', 'is', 'happy', 'but', 'the', 'cat']]
-    """
-    pass
 
+# получение конкорданса для word
+def get_concordance(tokens: list, word: str, left_n: int, right_n: int) -> list:
```

Comment:
Please, do not alter the interface of the given functions nor rename the arguments.

Reply:
okay, did not know. Changed